src/vcpkg/base/system.process.cpp

```diff
@@ -246,6 +251,100 @@ namespace vcpkg
+
+        p.skip_whitespace();
+        p.require_character('(');
+        auto executable = p.match_until([](char32_t c) { return c == ')'; }).to_string();
+        p.next();
+
+        p.skip_whitespace();
+        p.next(); // state
+
+        p.skip_whitespace();
+        auto ppid_str = p.match_while(ParserBase::is_ascii_digit);
+        auto maybe_ppid = Strings::strto<int>(ppid_str);
+        if (!maybe_ppid)
+        {
+            return nullopt;
+        }
+
+        return ProcessStat{maybe_ppid.value_or_exit(VCPKG_LINE_INFO), executable};
+    }
+
+    void get_parent_process_list(std::vector<std::string>& ret)
+    {
+#if defined(_WIN32)
+        // Enumerate all processes in the system snapshot.
+        auto snapshot = CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0);
+        if (snapshot == INVALID_HANDLE_VALUE) return;
+
+        std::map<DWORD, DWORD> pid_ppid_map;
+        std::map<DWORD, std::string> pid_exe_path_map;
+
+        PROCESSENTRY32W entry;
+        memset(&entry, 0, sizeof(entry));
+        entry.dwSize = sizeof(entry);
+        if (Process32FirstW(snapshot, &entry))
+        {
+            do
+            {
+                pid_ppid_map.emplace(entry.th32ProcessID, entry.th32ParentProcessID);
+                pid_exe_path_map.emplace(entry.th32ProcessID, Strings::to_utf8(entry.szExeFile));
+            } while (Process32NextW(snapshot, &entry));
+        }
+        CloseHandle(snapshot);
+
+        // Find hierarchy of current process
+        auto it = pid_ppid_map.find(GetCurrentProcessId());
+        if (it != pid_ppid_map.end())
+        {
+            // skip the vcpkg process
+            it = pid_ppid_map.find(it->second);
+            while (it != pid_ppid_map.end())
+            {
+                auto name = pid_exe_path_map[it->first];
+                ret.push_back(name);
+                it = pid_ppid_map.find(it->second);
+            }
+        }
+#elif defined(__linux__)
+        auto& fs = get_real_filesystem();
+        auto get_process_stat = [&](int pid) -> Optional<ProcessStat> {
+            const auto stat_file = fmt::format("/proc/{}/stat", pid);
+            if (!fs.exists(stat_file, IgnoreErrors{})) return nullopt;
+
+            std::error_code ec;
+            auto contents = fs.read_contents(stat_file, ec);
+            if (ec) return nullopt;
+
+            return try_parse_process_stat_file(contents, stat_file);
+        };
+
+        auto vcpkg_pid = getpid();
+        auto maybe_vcpkg_stat = get_process_stat(vcpkg_pid);
+        if (!maybe_vcpkg_stat) return;
+
+        auto vcpkg_stat = maybe_vcpkg_stat.value_or_exit(VCPKG_LINE_INFO);
+        auto pid = vcpkg_stat.ppid;
+        while (pid != 0)
+        {
+            auto maybe_stat = get_process_stat(pid);
+            if (!maybe_stat) break;
+
+            auto stat = maybe_stat.value_or_exit(VCPKG_LINE_INFO);
+            ret.push_back(stat.executable_name);
+            pid = stat.ppid;
```

Comment:
```suggestion
            if (auto stat = maybe_stat.get()) {
                ret.push_back(stat->executable_name);
                pid = stat->ppid;
            } else {
                break;
            }
```

This way you don't need assertions / VCPKG_LINE_INFO